### PR TITLE
Create a Proxy Metrics store + retrieval endpoint

### PR DESCRIPTION
**Pull request by spember**
## Description

This PR adds a few things to the Proxy:

* a centralized Observer, allowing a common place for the proxies to publish events and allow other Observers to act on them
* Created the first such Observer, a singleton Proxy Metrics store which tracks the total requests made, and any cache misses
* a special HTTP 1 endpoint which allows for retrieval of these metrics.

The data can be fetched via: `https://<proxy-ip>/rp-cache-info`. (This is naively assuming there's no other path in the world that starts with "rp-cache-info") It also supports a optional query parameter, `n`, mentioned below.

The output of the endpoint shows the totalRequests, hits, misses, and cache key size of the life span of the Proxy. It also includes a field 'topMissCounts' (the size of which is controlled by `n`, e.g. ?n=15), which lists the paths that had a cache miss and the number of occurrences of those misses.

### CODE REVIEW

File: packages/proxy/src/reverse-proxy.ts [hunk 1]
```diff
@@ -24,19 +45,24 @@ export const handleIncomingRequest = async ({
   cache: ProxyCache;
   handleUncached: Function;
 }) => {
+  if (request.url?.startsWith(CACHE_INFO_URL)) {
```
Comment (JasonPapanicholasWF) [inline packages/proxy/src/reverse-proxy.ts:48]: Think you can use path here instead of startsWith()

Comment (JasonPapanicholasWF) [reply packages/proxy/src/reverse-proxy.ts:48]: And you could also check if the host is `racer`

File: packages/proxy/src/index.ts [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 import {buildHttpReverseProxy, buildHttp2ReverseProxy} from './reverse-proxy';
+import {getProxyMetrics} from './proxy-metrics';
 
 const HTTP_DEFAULT_PORT = 80;
 const HTTPS_DEFAULT_PORT = 443;
 
 const runProxy = async () => {
   // Inject for testability
   const cache = new ProxyCache();
+  // in order to appropriately listen for events ocurring in both of these servers, let's create
```
Comment (JasonPapanicholasWF) [inline packages/proxy/src/index.ts]: occurring* :)

**Review by JasonPapanicholasWF — APPROVED** [review]
Couple comments

File: packages/proxy/src/reverse-proxy.ts (hunk 1 above)
Comment (spember) [reply packages/proxy/src/reverse-proxy.ts:48]: host is the target url, no? Or do you mean `proxy`? still, when hitting it from my terminal I need to hit by ip as it's outside of the docker network stack.

File: packages/proxy/src/index.ts (hunk 2 above)
Comment (spember) [reply packages/proxy/src/index.ts]: I can't spell that word. Also have trouble with 'separate'

**Pull request merged**